phoenix-core/src/main/java/org/apache/phoenix/mapreduce/AbstractBulkLoadTool.java

```diff
@@ -242,16 +252,31 @@ private int loadData(Configuration conf, CommandLine cmdLine) throws Exception {
 
         List<TargetTableRef> tablesToBeLoaded = new ArrayList<TargetTableRef>();
         PTable table = PhoenixRuntime.getTable(conn, qualifiedTableName);
+
+        boolean isImmutable = table.isImmutableRows();
```

Comment:
I think that bulk loading does not observe immutability, and you can modify rows with it.
If that's true, then we shouldn't treat immuable tables any different than normal ones.
Could you please check this ?